### ISSUE TITLE
Migrate away from arbitrary reference dates

### ISSUE THREAD

**Issue by sophiamersmann** [owid/owid-grapher#2140]
Unblocks #2007 

## Summary

**What:** Migrate away from specifying dates as the number of days since an arbitrary reference date that is either chosen by an author or defaulting to 2020-01-21
**Why:** To make internal date representation more sane (and also to prepare for [future work supporting more time intervals](https://www.notion.so/owid/2023-04-13-Support-for-more-time-intervals-in-Grapher-3b927a4d04aa4e46b480d5efb46f8410))
**How:** Substantial database migrations, plus some work in the Grapher codebase

## Current state

- Time points are represented as integers and either...
	- interpreted as years
	- interpreted as the number of days since a reference date (if  the metadata field `display.yearIsDay` is true)
		- the reference date is either taken from `display.zeroDay`
		- or the default reference date is used (`EPOCH_DATE`): 2020-01-21

## Concerns

- Specifying dates with reference to an arbitrary date in Jan 2020 is weird/unexpected
- Arbitrary reference dates add unnecessary complexity

## Actionable plan

**Step 1: Change the default reference date**
- [ ] Write a database migration: For all variables without an explicit `zeroDay`, set the `zeroDay` to the the current default date, 2020-01-21
- [ ] Change the `EPOCH_DATE` in Grapher to 1970-01-01

**Step 2: Clean up database and make sure all variables use the new default reference date (1970-01-01)**
- [ ] Write a database migration: For all variables with daily data, update the time column to use the new default reference date (1970-01-01)
- [ ] Remove `display.zeroDay`

**(optional) Step 3: Store years and dates in different columns**

- [ ] Add a new column `data_values.date`
- [ ] For all variables with daily data, move time points from `data_values.year` to `data_values.date`

**Comment by marcelgerber** [comment]
Makes sense!

PS: I changed `display.zeroDate` to `display.zeroDay` in the above text, which is what the field is actually called :)

**Comment by danyx23** [comment]
We talked abou this - it's probably a lot of work for relatively little benefit. It is probably better to concentrate on building a new, better solution for dealing with time as part of #2007 and then deprecate `yearIsDay` or migrate away from it if that is worthwhile.
